discovery_engine_core/core/src/config.rs

```diff
@@ -65,6 +65,10 @@ pub struct InitConfig {
     pub de_config: Option<String>,
     /// Log file path.
     pub log_file: Option<String>,
+    /// Directory in which user data should be stored.
+    pub data_dir: String,
+    /// Use a in-memory db instead of a db in the `data_dir`
+    pub use_in_memory_db: bool,
```

Comment:
❓ wdyt about merging both fields into an `Option<String>` which either implies in-memory db if `None` or a db directory if `Some(data_dir)`?

Reply:
I had considered this, but in the end this is the generic application data directory which is used for more things then just the database.  Using `Option<String>` would work currently as we only use it from rust for the db. But IMHO it would sooner or later brake with us moving more parts to rust. Which is why I kept it separate.